preview_generator/utils.py

```diff
@@ -155,3 +156,24 @@ def default(self, obj: typing.Any) -> str:
             return serial
 
         return JSONEncoder.default(self, obj)
+
+
+def get_decrypted_pdf(stream, strict=True, warndest=None, overwriteWarnings=True):
+    """
+    Return a PdfFileReader object decrypted with default empty key (if file is encrypted)
+    The signature is taken from PdfFileReader.__init__
+
+    See https://github.com/algoo/preview-generator/issues/52
+    which is related to https://github.com/mstamy2/PyPDF2/issues/51
+
+    :param stream:
+    :param strict:
+    :param warndest:
+    :param overwriteWarnings:
+    :return:
+    """
+    pdf = PdfFileReader(stream, strict, warndest, overwriteWarnings)
+    if pdf.isEncrypted:
+        pdf.decrypt('')
```

Comment:
What happened if pdf is encrypted with a real password ?

Reply:
Good question. The right way to manage this would be to allow to give a password in order to decrypt the PDF.

I suggest you add an issue about supporting password-protected PDF and we will manage it properly later.

At the moment, there are no regression: encrypted PDF were not supported at all.

Reply:
I added a TODO for that: https://github.com/algoo/preview-generator/pull/53/commits/eb355e0397a0c5feb673d3fd630b848d63ad04c8

Reply:
https://github.com/algoo/preview-generator/issues/54